test/plugin/test_out_google_cloud.rb

```diff
@@ -1295,140 +148,24 @@ def setup_logging_stubs
-                 headers: { 'Content-Length' => FAKE_AUTH_TOKEN.length,
-                            'Content-Type' => 'application/json' })
-
-    stub_request(:post, 'https://www.googleapis.com/oauth2/v3/token')
-      .with(body: hash_including(grant_type: 'refresh_token'))
-      .to_return(body: %({"access_token": "#{FAKE_AUTH_TOKEN}"}),
-                 status: 200,
-                 headers: { 'Content-Length' => FAKE_AUTH_TOKEN.length,
-                            'Content-Type' => 'application/json' })
-  end
-
-  def setup_managed_vm_metadata_stubs
-    stub_metadata_request(
-      'instance/attributes/',
-      "attribute1\ngae_backend_name\ngae_backend_version\nlast_attribute")
-    stub_metadata_request('instance/attributes/gae_backend_name',
-                          MANAGED_VM_BACKEND_NAME)
-    stub_metadata_request('instance/attributes/gae_backend_version',
-                          MANAGED_VM_BACKEND_VERSION)
-  end
-
-  def setup_container_metadata_stubs
-    stub_metadata_request(
-      'instance/attributes/',
-      "attribute1\nkube-env\nlast_attribute")
-    stub_metadata_request('instance/attributes/kube-env',
-                          "ENABLE_NODE_LOGGING: \"true\"\n"\
-                          'INSTANCE_PREFIX: '\
-                          "gke-#{CONTAINER_CLUSTER_NAME}-740fdafa\n"\
-                          'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
-  end
-
-  def setup_cloudfunctions_metadata_stubs
-    stub_metadata_request(
-      'instance/attributes/',
-      "attribute1\nkube-env\ngcf_region\nlast_attribute")
-    stub_metadata_request('instance/attributes/kube-env',
-                          "ENABLE_NODE_LOGGING: \"true\"\n"\
-                          'INSTANCE_PREFIX: '\
-                          "gke-#{CLOUDFUNCTIONS_CLUSTER_NAME}-740fdafa\n"\
-                          'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
-    stub_metadata_request('instance/attributes/gcf_region',
-                          CLOUDFUNCTIONS_REGION)
-  end
-
-  def container_log_entry_with_metadata(log)
-    {
-      log: log,
-      stream: CONTAINER_STREAM,
-      time: CONTAINER_TIMESTAMP,
-      kubernetes: {
-        namespace_id: CONTAINER_NAMESPACE_ID,
-        namespace_name: CONTAINER_NAMESPACE_NAME,
-        pod_id: CONTAINER_POD_ID,
-        pod_name: CONTAINER_POD_NAME,
-        container_name: CONTAINER_CONTAINER_NAME,
-        labels: {
-          CONTAINER_LABEL_KEY => CONTAINER_LABEL_VALUE
-        }
-      }
-    }
-  end
-
-  def container_log_entry(log, stream = CONTAINER_STREAM)
-    {
-      log: log,
-      stream: stream,
-      time: CONTAINER_TIMESTAMP
-    }
-  end
-
-  def cloudfunctions_log_entry(i)
-    {
-      stream: 'stdout',
-      log: '[D][2015-09-25T12:34:56.789Z][123-0] ' + log_entry(i)
-    }
+  def create_driver(conf = APPLICATION_DEFAULT_CONFIG, tag = 'test')
+    Fluent::Test::BufferedOutputTestDriver.new(
+      Fluent::GoogleCloudOutput, tag).configure(conf, use_v1_config: true)
```

Comment:
You might need to merge in Dhrupad's changes (#94) -- as I recall, he got rid of the erroneous `use_v1_config`, which is now flagged.